stregsystem/models.py

```diff
@@ -324,6 +329,81 @@ def delete(self, *args, **kwargs):
             super(Payment, self).delete(*args, **kwargs)
 
 
+class MobilePayment(models.Model):
+    UNSET = 'U'
+    APPROVED = 'A'
+    IGNORED = 'I'
+
+    STATUS_CHOICES = (
+        (UNSET, 'Unset'),
+        (APPROVED, 'Approved'),
+        (IGNORED, 'Ignored'),
+    )
+    member = models.ForeignKey(Member, on_delete=models.CASCADE, null=True,
+                               blank=True)  # nullable as mobile payment may not have match yet
+    payment = models.OneToOneField(Payment, on_delete=models.CASCADE, null=True, blank=True,
+                                   unique=True)  # Django does not consider null == null, so this works
+    customer_name = models.CharField(max_length=64)
+    timestamp = models.DateTimeField()
+    amount = models.IntegerField()
+    transaction_id = models.CharField(max_length=32,
+                                      unique=True)  # trans_ids are at most 17 chars, assumed to be unique
+    comment = models.CharField(max_length=128, blank=True, null=True)
+    status = models.CharField(max_length=1, choices=STATUS_CHOICES, default=UNSET)
+
+    def __str__(self):
+        return f"{self.member.username if self.member is not None else 'Not assigned'}, {self.customer_name}, " \
+               f"{self.timestamp}, {self.amount}, {self.transaction_id}, {self.comment}"
+
+    @transaction.atomic()
+    def delete(self, *args, **kwargs):
+        if self.id and self.payment is not None:
+            self.member.make_payment(-self.amount)
+            super(MobilePayment, self).delete(*args, **kwargs)
+            self.member.save()
+        else:
+            super(MobilePayment, self).delete(*args, **kwargs)
+
+    @staticmethod
+    @transaction.atomic
+    def submit_processed_mobile_payments(admin_user: User):
+        processed_payment: MobilePayment  # annotate iterated variable (PEP 526)
+        for processed_payment in make_processed_mobilepayment_query():
+
+            if processed_payment.status == MobilePayment.APPROVED:
+                payment = Payment(
+                    member=processed_payment.member,
+                    amount=processed_payment.amount)
+            elif processed_payment.status == MobilePayment.IGNORED:
+                payment = Payment(
+                    member=processed_payment.member,
+                    amount=0)
+            else:
+                raise RuntimeError("Trying to process MobilePayment not of status (APPROVED, IGNORED)")
```

Comment:
```suggestion
           else:
                payment = Payment(
                    member=processed_payment.member,
                    amount=0)
```
the function make_processed_mobilepayment_query makes this check. It can only return MobilePayment.IGNORED and MobilePayment.APPROVED :slightly_smiling_face:

Reply:
Good point, this check is just redundant :+1: